### PR TITLE
🚨 Refactorio #0: Fix all compiler warnings ♻️

### PR DESCRIPTION
**Pull request by jay-tux**
(as an added bonus, also removed some CLion warnings...)

### CODE REVIEW

File: src/Primitives/emulated.cpp [hunk 1]
```diff
@@ -365,7 +367,7 @@ def_prim(http_post, tenToOneU32) {
     uint32_t authorization = arg3.uint32;
     uint32_t authorization_len = arg2.uint32;
     int32_t response = arg1.uint32;
-    uint32_t size = arg0.uint32;
+    // uint32_t size = arg0.uint32; // never used in emulator
```
Comment (tolauwae) [inline src/Primitives/emulated.cpp:370]: we can just remove these lines

File: src/Debug/debugger.cpp [hunk 2]
```diff
@@ -341,35 +343,38 @@ bool Debugger::checkDebugMessages(Module *m, RunningState *program_state) {
 }
 
 // Private methods
-void Debugger::printValue(StackValue *v, uint32_t idx, bool end = false) const {
+void Debugger::printValue(const StackValue *v, const uint32_t idx,
+                          const bool end = false) const {
     char buff[256];
 
+#define FMT(fmt0) "%" fmt0
```
Comment (tolauwae) [inline src/Debug/debugger.cpp:350]: Did you verify this also works on the esp microcontrollers?

Comment (tolauwae) [reply src/Debug/debugger.cpp:350]: I checked it works :+1:

**Review by tolauwae — CHANGES_REQUESTED** [review]
Looks good. I would just delete the commented out lines.

File: src/Primitives/emulated.cpp (hunk 1 above)
Comment (MaartenS11) [reply src/Primitives/emulated.cpp:370]: The general idea was that if we needed them in the emulator in the future then all arguments would still be there and not just some of them with some gaps in between.

**Pull request merged**